osgar/bus.py

```diff
@@ -53,13 +54,20 @@ def __init__(self, logger, name):
         self.slots = {}
         self.stream_id = {}
         self._is_alive = True
+        self.no_output = set()
+        self.compressed_output = set()
 
     def register(self, *outputs):
-        for o in outputs:
+        for name_and_type in outputs:
+            o = name_and_type.split(':')[0]  # keep only prefixes
```

Comment:
Can we have `name_and_mod` as in modifier and split it to `o, mod` and later use `mod` instead of `.endswith()`?

Reply:
`mod` or `mode`? There does not have to be `:` so I am not sure how you mean it?